frontend/src/components/Connect/List/BatchActionsBar.tsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import {
+  Action,
+  ResourceType,
+  ConnectorAction,
+  FullConnectorInfo,
+} from 'generated-sources';
+import { Row } from '@tanstack/react-table';
+import { ActionButton } from 'components/common/ActionComponent';
+import { useConfirm } from 'lib/hooks/useConfirm';
+import {
+  useDeleteConnector,
+  useUpdateConnectorState,
+} from 'lib/hooks/api/kafkaConnect';
+import useAppParams from 'lib/hooks/useAppParams';
+import { ClusterNameRoute } from 'lib/paths';
+import { showServerError } from 'lib/errorHandling';
+
+interface BatchActionsBarProps {
+  rows: Row<FullConnectorInfo>[];
+  resetRowSelection(): void;
+}
+
+const BatchActionsBar: React.FC<BatchActionsBarProps> = ({
+  rows,
+  resetRowSelection,
+}) => {
+  const { clusterName } = useAppParams<ClusterNameRoute>();
+  const selectedConnectors = rows.map(({ original }) => original);
+  const stateMutation = useUpdateConnectorState(clusterName);
+  const deleteMutation = useDeleteConnector(clusterName);
+  const confirm = useConfirm();
+
+  const performBatchConnectorsAction = async (
+    confirmMessage: string,
+    action: ConnectorAction
+  ) => {
+    confirm(confirmMessage, async () => {
+      try {
+        await Promise.all(
+          selectedConnectors.map(({ connect, name }) =>
```

Comment:
@Haarolean can we have an endpoint for this that does bulk delete so we don't need to do this multiple promises here ?